# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1467294876 14400
#      Thu Jun 30 09:54:36 2016 -0400
# Node ID 1d775d40a6976b696f1d0ca2f958d5c447b4cd60
# Parent  f471f513ffd096dc6e24d7fabbb3264ad2dccab2
* mkbuiltins: require --source or --header option.

diff --git a/libinterp/mkbuiltins b/libinterp/mkbuiltins
--- a/libinterp/mkbuiltins
+++ b/libinterp/mkbuiltins
@@ -84,17 +84,17 @@ EOF
 
 #undef XDEFUN_DECL
 #undef XDEFUN_DLD_DECL
 
 #endif
 
 EOF
 
-else
+elif $make_source; then
 
   cat << \EOF
 // DO NOT EDIT!  Generated automatically by mkbuiltins.
 
 #if defined (HAVE_CONFIG_H)
 #include "config.h"
 #endif
 
@@ -178,11 +178,16 @@ EOF
 
   cat << \EOF
 
   install_built_in_docstrings ();
 }
 
 EOF
 
+else
+
+  echo "mkbuiltins: must specify --header or --source!" 1>&2
+  exit 1
+
 fi
 
 exit 0
